embedded-graphics/src/primitives/ellipse.rs

```diff
@@ -0,0 +1,459 @@
+    ///
+    /// ```
+    /// # use embedded_graphics::primitives::Ellipse;
+    /// # use embedded_graphics::prelude::*;
+    /// let mut ellipse = Ellipse::new(Point::new(5, 10), Size::new(10, 15));
+    /// ellipse.translate_mut(Point::new(10, 10));
+    ///
+    /// assert_eq!(ellipse.top_left, Point::new(15, 20));
+    /// ```
+    fn translate_mut(&mut self, by: Point) -> &mut Self {
+        self.top_left += by;
+
+        self
+    }
+}
+
+/// Iterator over all points inside the ellipse
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub struct Points {
+    ellipse: Ellipse,
+    iter: super::rectangle::Points,
+}
+
+impl Points {
+    fn new(ellipse: &Ellipse) -> Self {
+        Self {
+            ellipse: *ellipse,
+            iter: ellipse.bounding_box().points(),
+        }
+    }
+
+    fn empty() -> Self {
+        Self {
+            ellipse: Ellipse::new(Point::zero(), Size::zero()),
+            iter: Rectangle::new(Point::zero(), Size::zero()).points(),
+        }
+    }
+}
+
+impl Iterator for Points {
+    type Item = Point;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let ellipse = self.ellipse;
+
+        self.iter.find(|point| ellipse.contains(*point))
+    }
+}
+
+/// Pixel iterator for each pixel in the ellipse border
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub struct StyledEllipseIterator<C>
+where
+    C: PixelColor,
+{
+    iter: Points,
+    outer_size: Size,
+    outer_color: Option<C>,
+    inner_size: Size,
+    inner_color: Option<C>,
+    center: Point,
+    inner_width: u32,
+}
+
+impl<C> StyledEllipseIterator<C>
+where
+    C: PixelColor,
+{
+    fn new(styled: &Styled<Ellipse, PrimitiveStyle<C>>) -> Self {
+        // Always use a stroke width of 0 if no stroke color was set.
+        let stroke_width = styled.style.effective_stroke_width();
+
+        let outer_size = styled.primitive.size;
+        let inner_size = outer_size.saturating_sub(Size::new(2 * stroke_width, 2 * stroke_width));
+
+        let inner_width = inner_size.width;
+
+        let inner_size = Size::new(inner_size.width.pow(2), inner_size.height.pow(2));
+        let iter = if !styled.style.is_transparent() {
+            Points::new(&styled.primitive)
+        } else {
+            Points::empty()
+        };
+
+        Self {
+            iter,
+            outer_size,
+            outer_color: styled.style.stroke_color,
+            inner_size,
+            inner_color: styled.style.fill_color,
+            center: styled.primitive.center_2x(),
+            inner_width,
+        }
+    }
+}
+
+impl<C> Iterator for StyledEllipseIterator<C>
+where
+    C: PixelColor,
+{
+    type Item = Pixel<C>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let Self {
+            inner_size,
+            inner_color,
+            outer_color,
+            center,
+            inner_width,
+            ..
+        } = *self;
```

Comment:
I think this will copy more values than necessary per `next` call, which could impact performance. Using `for point in self.iter.by_ref() { ... }` isn't as nice as using `find_map` but you can use `self` inside the for loop.